learner.py

```diff
@@ -1,24 +1,38 @@
 import numpy as np
 from lib import *
 
+cost_power = 10
+
 def cost(Y, Yp):
-    return np.sum(np.square(Y-Yp))/2
+    return np.sum(np.power(Y-Yp, cost_power))/cost_power
+
+def cost_gradient(Y, Yp):
+    return np.power(Yp - Y, cost_power-1)
 
-def gradient_step(mp, X, Y, lrate):
+def gradient(mp, X, Y):
     sc = len(X)
     Yp, grad = mp.gradient(X)
-    step = np.sum((Yp - Y)*grad, axis=1)
+    step = np.sum(cost_gradient(Y, Yp)*grad, axis=1)
     step /= np.linalg.norm(step)
```

Comment:
Is normalization still appropriate?

Reply:
I'm not sure whether normalization was ever appropriate. I guess it smoothens areas of high gradient and plateaus.

Reply:
Removed